util/network-devp2p/src/host.rs

```diff
@@ -722,11 +725,24 @@ impl Host {
 							// Outgoing connections are allowed as long as their count is <= min_peers
 							// Incoming connections are allowed to take all of the max_peers reserve, or at most half of the slots.
 							let max_ingress = max(max_peers - min_peers, min_peers / 2);
-							if reserved_only ||
-								(s.info.originated && egress_count > min_peers) ||
-								(!s.info.originated && ingress_count > max_ingress) {
-								// only proceed if the connecting peer is reserved.
-								if !self.reserved_nodes.read().contains(&id) {
+							if reserved_only
+								|| (s.info.originated && egress_count > min_peers)
+								|| (!s.info.originated && ingress_count > max_ingress)
+							{
+								// We didn't start the connection, but the node is known to us
+								// So eventually we will attempt to connect to it as well.
```

Comment:
Have we checked that this is not subject to a variant on the table poisoning attack?

Let's say if there are N nodes controlled by the attacker in a client's node table. For outgoing connections, not all of those N nodes will be tried immediately. However, with this change, in incoming connections, the attacker can flood the client using all those N nodes, and controls a much larger portion of the ongoing connections.